tsdb/engine/tsm1/writer.go

```diff
@@ -463,6 +467,18 @@ func (d *directIndex) Close() error {
 	return os.Remove(d.fd.Name())
 }
 
+// Remove removes the index from any tempory storage
+func (d *directIndex) Remove() error {
+	if d.fd == nil {
+		return nil
+	}
+
+	if err := d.fd.Close(); err != nil {
```

Comment:
So if there is a problem closing the file we're not going to clean up, but we're not going to return an error to the caller either? This scenario seems unusual enough that we don't want to ignore it or not log it.

Reply:
That should be returning the `err` here.  Will fix.